uitools/cpp/Esri/ArcGISRuntime/Toolkit/BasemapGalleryController.cpp

```diff
@@ -99,73 +72,62 @@ namespace Toolkit {
+                       [self, listenToLoadSignals, mapOrScene](Basemap* oldBasemap)
                        {
                          QObject::disconnect(self, nullptr, oldBasemap, nullptr);
                          listenToLoadSignals(); // Connect to new basemap.
-                         emit self->currentBasemapChanged();
+                         self->setCurrentBasemap(mapOrScene->basemap());
                        });
 
       listenToLoadSignals();
     }
 
     /*!
       \internal
-      Connect to [Scene/Map]View and associated Map/Scene and associated Basemap.
+      Connect to [Scene/Map].
 
-      1. We Connect up to the current map/scene.
-      2. If the map/scene changes we disconnect from the old one.
-      3. We connect up to the new map/scene.
-      4. We emit a basemapChanged signal whenever the map or scene changes.
+      1. Update our cached basemap with the Scene/Map basemap.
+      1. Discover the runtime type of GeoModel
+      2. Connect to that type's basemapChanged signal.
      */
-    template <typename T, typename F>
-    void connectToView(BasemapGalleryController* self, T* view, F mapOrSceneChangedSignal)
+    void connectToGeoModel(BasemapGalleryController* self, GeoModel* geoModel)
     {
-      static_assert(std::is_same<T, SceneViewToolkit>::value || std::is_same<T, MapViewToolkit>::value, "Type must be a MapView or SceneView.");
-
-      auto item = getMapOrScene(view);
-      connectToMapScene(self, item);
-
-      // If the current map/scene changes, disconnect all connections to the old ones.
-      auto oldItem = qPointerFrom(item);
-      QObject::connect(view, mapOrSceneChangedSignal, self,
-                       [oldItem, self, view]
-                       {
-                         if (oldItem)
+      if (geoModel->loadStatus() == LoadStatus::Loaded)
+      {
+        self->setCurrentBasemap(geoModel->basemap());
+      }
+      else
+      {
+        QObject::connect(geoModel, &GeoModel::doneLoading, self, [self, geoModel](Error e)
                          {
-                           QObject::disconnect(oldItem, nullptr, self, nullptr);
-                           if (Basemap* basemap = oldItem->basemap())
+                           if (e.isEmpty())
                            {
-                             QObject::disconnect(basemap, nullptr, self, nullptr);
+                             self->setCurrentBasemap(geoModel->basemap());
                            }
-                         }
-                         // Connect up to new signals.
-                         connectToMapScene(self, getMapOrScene(view));
-                         emit self->currentBasemapChanged();
-                       });
+                         });
+      }
+
+      if (auto scene = qobject_cast<Scene*>(geoModel))
+      {
+        connectToBasemap(self, scene);
+      }
+      else if (auto map = qobject_cast<Map*>(geoModel))
+      {
+        connectToBasemap(self, map);
+      }
```

Comment:
Make a note. This can likely be cleaned up next release once the basemapChanged signal is moved to GeoModel.